src/System.Private.CoreLib/shared/System/Marvin.cs

```diff
@@ -28,73 +28,194 @@ internal static partial class Marvin
         /// </summary>
         public static int ComputeHash32(ref byte data, int count, uint p0, uint p1)
         {
-            nuint ucount = (nuint)count;
-            nuint byteOffset = 0;
+            // Control flow of this method generally flows top-to-bottom, trying to
```

Comment:
I wonder if there are some things we could do in JIT to make the code generate to what you wrote - it seems the code is now fairly complicated comparing to what it was